API.md

```diff
@@ -0,0 +1,136 @@
+# Mail Ballot Manager APIs
+
+## Open Questions
+
+- If there is an error in the election data, how does a election official start
+  over?
+- If there is an error in the voter mailing list, how does an election official
+  fix it?
+- If an election official needs to stop the presses, can they?
+
+## Types
+
+Election as defined in `@votingworks/ballot-encoder` and the following…
+
+```
+interface VoterMailBallot {
+  id: string,
+  firstName: string,
+  middleName: string,
+  lastName: string,
+  nameSuffix: string,
+  street1: string,
+  street2: string,
+  city: string,
+  state: string,
+  zipcode: string,
+  ballotStyle: string,
+  precinctId: string,
+  ballotfile: string, // filepath?
+  ballotCreated: timestamp,
+  ballotPrinterReceived: timestamp,
+  ballotPrinterPrinted: timestamp
+  ballotOutboundSent: timestamp
+  ballotOutboundDelivered: timestamp
+  ballotInboundSent: timestamp
+  ballotInboundDelivered: timestamp
+}
+```
+
+```
+interface BallotTemplate {
+  fileName: string,
+  fileContent: pdfFile,     // or filepath? Perhaps different for upload vs list.
+  ballotStyle: BallotStyle, // As as defined in `@votingworks/ballot-encoder`
+  precinct: Precinct,       // As as defined in `@votingworks/ballot-encoder`
+}
+```
+
+```
+interface InsertsData {
+  affadavit: string,
+  instructions: string,
+  helpPhone: string,
+  helpEmail: string,
+  helpWeb: string,
+  helpAddress: string,
+}
+```
+
+## Auth API
+
+Whatever Auth0 uses here.
+
+## Users API
+
+What user date is provided by default via Auth0?
+
+### `GET /auth/me`
+
+```
+{
+  id: string,
+  email: string,
+}
+```
+
+## Elections API
+
+### `GET /elections/list`
+
+```
+[{
+  id: string,
+  definition: Election,
+  packageHash: string,
+  printingAndMailingApprovedAt: datetime,
+  printingAndMailingApprovedBy: userId,
+  ballotTemplates: BallotTemplate[],
+  voters: VoterMailBallot[],
```

Comment:
Is this all voters for an election? This is almost certainly going to be huge, right?

Reply:
Yes. Wasn't sure how to model batches, so I figured I'd just keep it simple and add 'em all to start. 

I'd like to define what exactly is huge? How many records? I've heard that React can handle a lot of data and that how much is rendered is the part that should be optimized for speed. Looking forward to learning and determining these unknowns.

Reply:
For simplicity's sake, sure. I'm guessing on the order of 10K would be pretty comfortable. More than that and I'd start feeling anxious.

Reply:
As for how to model batches, we'd want another endpoint like this:

```
$ curl -vv https://mail-ballot-manager.votingworks.app/election/abc123/voters
…
HTTP/1.1 200 OK
Content-Type: application/json
X-Next-Page: https://mail-ballot-manager.votingworks.app/election/abc123/voters?start=def456

[
  {
    "id": …
  },
  …
]
```